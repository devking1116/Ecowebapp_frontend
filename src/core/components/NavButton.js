/* @flow */
import React from 'react';
import { Button } from 'semantic-ui-react';

type Props = {
  children?: any,
};

/**
 * Component that works as a button for the navigation bar.
 */
export default function NavButton({ children, ...otherProps }: Props): React.Element<*> {
  return (
    <Button color="blue" {...otherProps}>
      {children}
    </Button>
  );
}
