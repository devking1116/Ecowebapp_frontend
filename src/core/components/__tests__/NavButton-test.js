import React from 'react';
import renderer from 'react-test-renderer';

import NavButton from '../NavButton';

test('it displays its children as the label of the item.', () => {
  const component = renderer.create(
    <NavButton>
      Test Child
    </NavButton>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
