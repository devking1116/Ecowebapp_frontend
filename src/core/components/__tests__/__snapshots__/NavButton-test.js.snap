exports[`test it displays its children as the label of the item. 1`] = `
<button
  className="ui blue button"
  onClick={[Function]}
  tabIndex={undefined}>
  Test Child
</button>
`;
